File: scripts/dmzMBRAPluginLayerActive.lua
local ActiveHandle = dmz.handle.new ("Layer_Active")

function update_object_flag (self, handle, attr, value)
   if value then
      if self.active and self.active ~= handle then
         dmz.object.flag (self.active, ActiveHandle, false)
      end
      self.active = handle
   else
      if self.active == handle then self.active = nil end
   end
end

function new (config, name)

   local self = {
      log = dmz.log.new ("lua." .. name),
      objObs = dmz.object_observer.new (),
   }

   self.log:info ("Creating plugin: " .. name)

   local cb = { update_object_flag = update_object_flag }
   self.objObs:register (ActiveHandle, cb, self)

   return self
end

File: scripts/dmzMBRAPluginLayerActive.js
var dmz =
   { defs: require("dmz/runtime/definitions")
   , object: require("dmz/components/object")
   }
   , ActiveHandle = dmz.defs.createNamedHandle ("Layer_Active")
   , active = null
   ;


dmz.object.flag.observe (self, ActiveHandle, function (handle, attr, value) {
   if (value) {
      if (active && active != handle) {
         dmz.object.flag (active, ActiveHandle, false);
      }
      active = handle;
   } else if (active == handle) { active = null; }
});
